library/microbenchmarks/gpu/gpu_burn/__init__.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+__all__ = ['GpuBurnBase']
+
+
+class GpuBurnBase(rfm.RegressionTest, pin_prefix=True):
+    '''Base class for the GPU Burn test.
+
+    -- Compile --
+    The tests sources can be compiled for both CUDA and HIP. This is set with
+    the `gpu_build` variable, which must be set by a derived class. This
+    source code can also be compiled for a specific device architecture by
+    defining the `gpu_arch` variable.
+
+    -- Run --
+    The variables required by the run stage are:
+     - executable_opts: Defines the options to be passed to the executable.
+     - num_tasks: Number of tasks to use for this test.
+
+    -- Sanity --
+    Checks that the output matches the number of num_gpus_per_node*num_tasks.
+
+    -- Performance --
+    The performance patterns are:
+     - perf: Gflops/s achieved on average for the duration of the run.
+     - temp: the GPU's temperature in degC after the run.
```

Comment:
The maximum of all nodes.

Reply:
Performance library tests should also define a dummy reference just to set the units. This will cause ReFrame to automatically show the units in its run report as well as with the `--performance-report` option. You can achieve this by simply setting:

```python
reference = {
    '*': {
        'perf': (0, None, None, 'Gflop/s'),
        'temp': (0, None, None, 'degC')
    }
}
```

Reply:
I would even add this always as a pre-performance hook. This will allow users to define references partially. Otherwise, ReFrame complains about unresolved references.

Reply:
I think I like more the option of setting it into the class body. Setting the `references` with a hook would prevent any of the derived classes from setting the references directly in the class body; unless of course you disable the hook. But that just seems like a lot more work than trying to reuse these 2 lines of code from the dummy references 😂

Reply:
Perhaps we need a feature to set the units of the performance variables without defining the `reference`. Let me think a bit about it.

Reply:
Took me a while to find it, but I was convinced it was not the first time I had heard that! 😂 See #1568